payloads/start.go

```diff
@@ -242,6 +242,10 @@ type StartCmd struct {
 	// Storage contains all the information required to attach or boot
 	// from storage for the new instance.
 	Storage []StorageResource `yaml:"storage,omitempty"`
+
+	// Migration is set to true if the payload represents a request to
+	// restart an existing instance on a new node.
+	Migration bool
```

Comment:
> although in some ways I'm already breaking the existing model by explicitly setting the state of instances inside controller.

Thinking about this a bit more I don't think this PR does violate controller's current approach to instance state, in which an instances state is something that is owned by launcher.  It's true that this PR sets the instance state to Stopped but this is done as a response to an event received from launcher.  It could just as easily be done via STATS.

Getting rid of the Migration flags and InstanceStopped, introducing extra instance states in controller and having controller decide an instance's state (e.g., having it ignore instance states received in STATS from launcher) won't work in the current implementation as state is stored in /tmp.  A controller node reboot could result in the accidental deletion of a instance that the user restarted prior to the reboot.

That said the implementation in this PR has a state bug.  Stop an instance and reboot controller.  After the reboot, the instance's set is forever pending and it cannot be restarted.  The instance is no longer present on the compute node so launcher won't send any more stats for it.  This could be fixed by setting the status of instances that have not reported in for a fixed period of time to exited.  This would also fix the existing issue https://github.com/01org/ciao/issues/1038, providing the node doesn't come back up.

Moving instance state to permanent storage would also fix these problems.

I'm happy with either option, although I'm not sure how much of this work should be done as part of this PR.  @kaccardi do you have a preference or indeed a better way of solving these problems?